addons/mail/static/src/services/chat_window_service/chat_window_service.js

```diff
@@ -57,7 +55,6 @@ const ChatWindowService = AbstractService.extend({
             this.component = undefined;
         }
         const ChatWindowManagerComponent = components.ChatWindowManager;
-        ChatWindowManagerComponent.env = this.env;
```

Comment:
Yes, I think that the management of `env` in Odoo repository is rubish and does not allow anyone to use `env` the OWL intended way.

I had to remove these lines, because tests would create `env` later (another object) and these components would never collect messaging stuffs at all. In fact it would crash on invoking their `useStore()` because `env` has no `store`...

Anyway, I think we should think of `env` as unique and global env in our refactoring, that's why this solution is not the worst in my opinion...